packages/amplify_auth_plugin_interface/lib/src/Session/UpdateUserAttributesRequest.dart

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/foundation.dart';
+import 'AuthUserAttribute.dart';
+
+/// Encapsulates parameters for a update user attributes operation
+class UpdateUserAttributesRequest {
+  /// The list of user attribute to update
+  final List<AuthUserAttribute> attributes;
+
+  // ignore: public_member_api_docs
```

Comment:
I updated this and `serializeAsMap`.

I could go either way on something like this. On one hand, this isn't meant for public consumption (no harm in it being public, customers just shouldn't need to use it directly) and I don't really want to set a standard of adding documentation that isn't valuable. On the other hand, it does look make the API docs look incomplete to have so many public things undocumented.

Also worth noting that we do lose 10 points on pub.dev for having less than 20% of public APIs documented. I would assume simply adding ignore does not exclude this from that calculation.

Reply:
Fwiw, I actually think if it's meant to be internal, we should maybe emphasize that it's not meant to be publicly consumed - i.e. it's subject to change so use at your own risk. I actually wonder if we would benefit from doing something like that across the board.

Generally speaking, I don't like to add documentation that isn't all that meaningful either but if I consider the developer experience it seems like, at worst, extra documentation is annoying but, at best, really helps me navigate foreign code. Since our code is largely intended to be consumed by others, even seemingly low value documentation could be surprisingly helpful... for example, if someone didn't understand Dart at all and was looking at the constructor syntax for the first time, they may find it useful to know the constructor is the constructor.

Anyway, sorry for the long winded way of saying thanks for adding the doc 😂